nikola/plugin_categories.py

```diff
@@ -338,7 +338,12 @@ def split_metadata(self, data):
         This splits in the first empty line that is NOT at the beginning
         of the document.
         """
-        split_result = re.split('(\n\n|\r\n\r\n)', data.lstrip(), maxsplit=1)
+        if data.startswith('---'):  # YAML metadata
+            split_result = re.split('(---\n\n|----\r\n\r\n)', data.lstrip(), maxsplit=1)
```

Comment:
Also: isn't it better to split along `\n\n---\n\n` (for Unix-style newlines) instead of `---\n\n`? Otherwise, a line ending with `---` will also suffice for a split (which is probably not what you want), as opposed to a line containing only `---`.